WIP: replace fail2ban by reaction

ppom0 opened the pull request:
## The problem

fail2ban

## Solution

reaction

[introduction to problem & solution](https://blog.ppom.me/en-reaction/)

## PR Status

Very much draft

## How to test

Absolutely no migration ready for now, I'm just opening this to let us know that it's a WIP.

Commit:
Add postfix-sasl filter

Commit:
WIP reaction configuration

Commit:
Add recidive stream

Commit:
Add ip pattern configuration

Thovi98 commented:
This seems very nice, thanks!
I’ve just a little observation: fail2ban has a great community, while reaction seems to be only maintained by you (If I’m not mistaken). What is the reliability of reaction being maintained if you decide to stop working on it for example? :)

Commit:
tail --quiet

zamentur commented:
Related to https://github.com/YunoHost/issues/issues/2423

Commit:
add ssh regex

Commit:
First draft of a reaction helper

Commit:
Add bash shebang

Commit:
Add a one-week ban for recidive

Commit:
YOLO edits to the reaction helper

What about applications that would log to journald?
What about different retry/retryperiod options?

Commit:
Use the virtual plugin instead of custom file based stream.

Commit:
Ban command more readable

Commit:
Add PAM auth stream+filter

Commit:
Merge mail streams. More regexes

Commit:
postfix: Fix ban call

Commit:
small changes

Commit:
Merge nginx things

ppom0 commented:
When fail2ban has an invalid jail or filter file, it seems to simply ignore it and mark the jail as failed.

The fail2ban_ynh helper detects it and logs a warning.

---

When reaction has an invalid config file, it refuses to start.

I'm going to do this for now: suffix the filename with `.failed` so that it is ignored by reaction, restart reaction, and log a warning.

---

Tell me if you'd like the helper to handle the error in another way (ie. cancel the app install, like the nginx_ynh helper...)

DeMiro5001 commented:
I am really bad at regex, in most of the packages I have created, I mostly skip fail2ban.
Plus I have noticed that postfix is not correctly configured in fail2ban that is reported in a dedicated issue. Sometimes one ip has more than 100 failed authentication attempts in 24h on one of my server. So I asked an AI (😶‍🌫️) to fix the postfix jail and I am currently testing it (will have more info in the next few days).
So, as a packager, how will this be in reaction? How can we test the config ? 
Does reaction have a webui?
I currently have a small script that sends me daily reports about fail2ban : banned ips, geolocation, etc..

ppom0 commented:
I won't answer specifically for postfix because I don't know :)

The packaging shouldn't change for reaction. I just rewrote the ynh_config_add_fail2ban helper so that it works transparently for reaction.

It's possible to test the config too, something close to `fail2ban-regex`.

reaction doesn't have a webui, but it has a CLI to see what's unbanned, and unban IPs.

I think changing a few lines in your script should be enough :)
